cpp/src/arrow/memory_pool_jemalloc.cc

```diff
@@ -153,4 +154,59 @@ Status jemalloc_set_decay_ms(int ms) {
 
 #undef RETURN_IF_JEMALLOC_ERROR
 
+#ifdef ARROW_JEMALLOC
```

Comment:
I don't think the ifdef is necessary? This file will only be compiled if jemalloc is enabled.

Reply:
Indeed! Removing.